### DISCUSSION TITLE
[projects beta] Ability to create an iteration with arbitrary date

### DISCUSSION THREAD

**Discussion by kineticac** [community/community#6494] [Projects and Issues]
Sometimes we need to put in a patch in the middle of an iteration that can overlap with our current cadence. Maybe this is bad practice, but there are occasions when this is required.

Iterations currently cannot overlap and you can't go back to create one.

It would be great to create an iteration by picking a start/end date.

**Comment by wdanilo** [comment]
Yes, lack of this just makes iterations completely unusable for us. Forgetting to set up iteration one day causes all subsequent iterations shifted. Being able to set up a start day is a must have.

**Reply by kineticac** [reply]
We ended up creating a "patch" attribute that has 1-off weekly patches. It still shows on kanban/table views, and you can still sort by that. It's not ideal, but it works for now.

**Comment by despairblue** [comment]
I started using a number field instead. 

Where iterations keep on counting you can then put in `1.5` if you want to do something in between.

<table>
<tr>
	<td>Iteration
	<td>Sprint (Number)
<tr>
	<td>1
	<td>1
<tr>
	<td>2
	<td>2
<tr>
	<td>3
	<td>2.5
<tr>
	<td>4
	<td>3
</table>

Works for us so far.

**Comment by brylie** [comment]
Another use case is for holidays. E.g., we don't want to plan an iteration over the Christmas break. So, allowing an arbitrary start date would be helpful so we can skip some weeks. Unfortunately, the iterations feature doesn't currently allow us to omit any days.

![Peek 2021-12-14 13-19](https://user-images.githubusercontent.com/17307/145988777-d7a4f4a5-1741-4dff-9bb7-f8b8499a4f07.gif)

Similarly, if I start an iteration on a Tuesday, the inability to skip any timespan or specify the start date means that all subsequent iterations must also begin on Tuesdays.

**Comment by azenMatt** [comment]
We just released some updates - https://github.blog/changelog/2022-02-02-the-new-github-issues-02-02-update/ - that make it easier to change the dates of iterations. Does this help solve these problems for you? 

However, this still doesn't allow overlapping - but should solve some of the difficulties in needing to move an iteration around after it has been created

**Reply by brylie** [reply]
It works better now. I'll test it out to make sure our iterations can skip a week ASAP.

**Comment by Potherca** [comment]
Although the "break" feature is great, we have a usecase that still needs an "in between" iteration.
We need to abandon the current sprint because of a shift in the short-term priorities.
Everthing else remains the same (issues assigned to sprints, etc.) we just need an iteration in between two existing ones (where there is now a "break").

Either being able to assign issues to that break or being able to add an interation in between two existing ones (after changing the dates, so there is no overlap) would be of great help!

Also related to: https://github.com/community/community/discussions/32136

**Comment by DNin01** [comment]
Yeah, this is one of Milestones' advantages. I wish iterations covered a couple more potential scheduling strategies; as they are now, iterations are kind of made for linear schedules with deadlines.

**Comment by msdisme** [comment]
I am trying to create trainings and the inability to set iterations in the past makes this really hard.

**Comment by artur-klesun-st** [comment]
@azenMatt would it be possible to make it possible to create iterations for past dates? It's very hard to maintain historical project data without being able to add missing records for past dates to correct and inconsistency. The only way right now is to delete ALL iterations following the date I want to add and re-create every single one of them from scratch...
https://github.com/orgs/community/discussions/32136
